src/View/Helper/TimeHelper.php

```diff
@@ -269,12 +269,9 @@ public function timeAgoInWords($dateTime, array $options = []): string
             'timezone' => null,
         ];
         $options['timezone'] = $this->_getTimezone($options['timezone']);
-        if (
-            $options['timezone'] &&
-            ($dateTime instanceof Time ||
-            $dateTime instanceof FrozenTime)
-        ) {
-            $dateTime = $dateTime->timezone($options['timezone']);
+        /** @psalm-suppress UndefinedInterfaceMethod */
+        if ($options['timezone'] && $dateTime instanceof DateTimeInterface) {
+            $dateTime = $dateTime->setTimezone($options['timezone']);
```

Comment:
Should we be suppressing this warning? Would it be better to use method_exists here as there is no good interface type to use.

Reply:
I want with practicality over technical correctness here.

Both DateTime and DateTimeImmutable has setTimezone() but sadly it's not part of DatetimeInterface. I really don't see someone coming up with a new implementation of DatetimeInterface which doesn't have setTimezone().